validate_sample.py

```diff
@@ -0,0 +1,399 @@
+    column = 'LAD'
+    maximum_length = 9
+    mandatory = True
+    if _check_column_exists(count, column, mandatory, sample_row):
+        lad = sample_row[column]
+        _check_length(column, lad, count, maximum_length)
+
+
+def validate_region(count, sample_row):
+    column = 'REGION'
+    maximum_length = 9
+    mandatory = True
+    if _check_column_exists(count, column, mandatory, sample_row):
+        region = sample_row[column]
+        _check_length(column, region, count, maximum_length)
+
+
+def validate_htc_willingness(count, sample_row):
+    global error_count
+    column = 'HTC_WILLINGNESS'
+    mandatory = True
+    if _check_column_exists(count, column, mandatory, sample_row):
+        htc_willingness = sample_row[column]
+        if htc_willingness.isnumeric and len(htc_willingness) == 1:
+            return
+        print(f'Line {count}: {column} {htc_willingness} is not valid.')
+        error_count += 1
+
+
+def validate_htc_digital(count, sample_row):
+    global error_count
+    column = 'HTC_DIGITAL'
+    mandatory = True
+    if _check_column_exists(count, column, mandatory, sample_row):
+        htc_digital = sample_row[column]
+        if htc_digital.isnumeric and len(htc_digital) == 1:
+            return
+        print(f'Line {count}: {column} {htc_digital} is not valid.')
+        error_count += 1
+
+
+def validate_fieldcordinator_id(count, sample_row):
+    column = 'FIELDCOORDINATOR_ID'
+    maximum_length = 7
+    mandatory = False
+    if _check_column_exists(count, column, mandatory, sample_row):
+        fieldcordinator_id = sample_row[column]
+        _check_length(column, fieldcordinator_id, count, maximum_length)
+
+
+def validate_fieldofficer_id(count, sample_row):
+    column = 'FIELDOFFICER_ID'
+    maximum_length = 10
+    mandatory = False
+    if _check_column_exists(count, column, mandatory, sample_row):
+        fieldofficer_id = sample_row[column]
+        _check_length(column, fieldofficer_id, count, maximum_length)
+
+
+def validate_treatment_code(count, sample_row):
+    column = 'TREATMENT_CODE'
+    maximum_length = 10
+    mandatory = True
+    if _check_column_exists(count, column, mandatory, sample_row):
+        treatment_code = sample_row[column]
+        _check_length(column, treatment_code, count, maximum_length)
+        _is_valid_treatment_code(count, treatment_code)
+
+
+def validate_ce_expected_capacity(count, sample_row):
+    global error_count
+    column = 'CE_EXPECTED_CAPACITY'
+    maximum_length = 4
+    mandatory = False
+    if _check_column_exists(count, column, mandatory, sample_row):
+        value = sample_row[column]
+        _check_length(column, value, count, maximum_length)
+        if not value == "" and not value.isnumeric():
+            print(f'Line {count}: {column}: {value} is not a valid integer.')
+            error_count += 1
+
+
+def _check_column_exists(count, column, mandatory, sample_row):
+    global error_count
+    if column not in sample_row:
```

Comment:
This should be checked in the header validation rather than on every row.

Reply:
Amended to only check that a value exists of the field is mandatory, as the presence of a column is already checked in the header check.